apps/extension/src/ui/domains/Settings/ManageNetworks/NetworkForm/SubNetworkForm.tsx

```diff
@@ -0,0 +1,391 @@
+import { useIsBuiltInChain } from "@ui/hooks/useIsBuiltInChain"
+import { useSetting } from "@ui/hooks/useSettings"
+import useToken from "@ui/hooks/useToken"
+import { isCustomChain } from "@ui/util/isCustomChain"
+import { ReactNode, useCallback, useEffect, useMemo, useRef, useState } from "react"
+import { FormProvider, useForm } from "react-hook-form"
+import { Trans, useTranslation } from "react-i18next"
+import { useDebounce } from "react-use"
+import { Button, Checkbox, Dropdown, FormFieldContainer, FormFieldInputText } from "talisman-ui"
+
+import { getSubNetworkFormSchema } from "./getSubNetworkFormSchema"
+import { getSubstrateRpcInfo } from "./helpers"
+import { NetworkRpcsListField } from "./NetworkRpcsListField"
+import { RemoveSubNetworkButton } from "./RemoveSubNetworkButton"
+import { ResetSubNetworkButton } from "./ResetSubNetworkButton"
+
+type SubNetworkFormProps = {
+  chainId?: ChainId
+  onSubmitted?: () => void
+}
+
+export const SubNetworkForm = ({ chainId, onSubmitted }: SubNetworkFormProps) => {
+  const { t } = useTranslation("admin")
+  const isBuiltInChain = useIsBuiltInChain(chainId)
+
+  const { chains } = useChains(true)
+  const [useTestnets, setUseTestNets] = useSetting("useTestnets")
+
+  const { defaultValues, isCustom, isEditMode, chain } = useEditMode(chainId)
+
+  const schema = useMemo(
+    () => getSubNetworkFormSchema(chain?.genesisHash ?? undefined),
+    [chain?.genesisHash]
+  )
+
+  // because of the RPC checks, do not validate on each change
+  const formProps = useForm<RequestUpsertCustomChain>({
+    mode: "onBlur",
+    defaultValues,
+    resolver: yupResolver(schema),
+  })
+
+  const {
+    register,
+    handleSubmit,
+    setValue,
+    watch,
+    resetField,
+    clearErrors,
+    setError,
+    reset,
+    trigger,
+    formState: { errors, isValid, isSubmitting, isDirty, touchedFields },
+  } = formProps
+
+  const { rpcs, id, nativeTokenCoingeckoId, accountFormat } = watch()
+
+  // initialize form with existing values (edit mode), only once
+  const initialized = useRef(false)
+  useEffect(() => {
+    if (chainId && defaultValues && !initialized.current) {
+      reset(defaultValues)
+      trigger("rpcs")
+      initialized.current = true
+    }
+  }, [defaultValues, chainId, reset, trigger])
+
+  // auto detect genesis hash (and token deets) based on RPC url (add mode only)
+  const rpcInfo = useRpcInfo(rpcs?.[0]?.url)
+  useEffect(() => {
+    if (chainId || !rpcInfo.isFetched) return
+    const rpcId = rpcInfo.data && `custom-${rpcInfo.data.genesisHash}`
+    if (rpcInfo.data && rpcId && rpcId !== id) {
+      setValue("id", rpcId, { shouldValidate: true, shouldTouch: true, shouldDirty: true })
+      setValue("genesisHash", rpcInfo.data.genesisHash as `0x${string}`, {
+        shouldValidate: true,
+        shouldTouch: true,
+        shouldDirty: true,
+      })
+      setValue("nativeTokenSymbol", rpcInfo.data.token.symbol, {
```

Comment:
detection of token name didn't work for me on a local polkadot chain (`./polkadot --dev`), seems blocking as it's not editable

Reply:
err, it's from aug 20th

Reply:
*we resolved this in chat*